strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java

```diff
@@ -126,8 +127,10 @@ public RepositoryPath resolve(Path other)
         {
             return this;
         }
-        
+
         other = unwrap(other);
+
+        validatePathRelativized(other);
```

Comment:
Why did you decide to go away from the proxying target path? 
isn't it was much more reliable and interesting? 
for me it was :)

Reply:
Heh :) thanks for you interest, @sbespalov 

Yeah, I have been trying to use proxied target path, everything looked quite straightforward when I thought everything is ready. Then hundreds of tests started failing. There were many issues out there and my next "fixes" started to grow up and I started feeling that everything has become overcomplicated. A turning point that led to my decision to give up was this method: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/nio/fs/WindowsPath.java#WindowsPath.toWindowsPath%28java.nio.file.Path%29 and generally `WindowsPath` doesn't want to deal with any other `Path` implementation than `WindowsPath` itself. "But hey, I want to use Proxy here and there" ;-) `ProviderMismatchException` killed my creativity :-)

Reply:
yeah, I see
you would have the same for Linux, I had to foresee this :(
sadness :(